en/02_Developer_Guides/00_Model/11_Scaffolding.md

```diff
@@ -88,34 +87,137 @@ class MyDataObject extends DataObject
 }
 ```
 
+### General search field
+
+Tabular views such as `GridField` or `ModalAdmin` include a search bar. As of Silverstripe CMS 4.12, the search bar will search across all of your searchable fields by default. It will return a match if the search terms appear in any of the searchable fields.
+
+#### Exclude fields from the general search
+
+If you have fields which you do _not_ want to be searched with this general search (e.g. date fields which need special consideration), you can mark them as being explicitly excluded by setting `general` to false in the searchable fields configuration for that field:
+
+```php
+use SilverStripe\ORM\DataObject;
+
+class MyDataObject extends DataObject
+{
+    private static $searchable_fields = [
+        'Name',
+        'BirthDate' => [
+            'general' => false
+        ],
+    ];
+}
+```
+
+#### Customise the general search field name
+
+By default the general search field uses the name "q". If you already use that field name or search query in your [SearchContext](/developer_guides/search/searchcontext), you can change this to whatever name you prefer either globally or per class:
+
+[hint]
+If you set `general_search_field_name` to any empty string, general search will be disabled entirely. Instead, the first field in your searchable fields configuration will be used, which was the default behaviour prior to Silverstripe CMS 4.12.
+[/hint]
+
+**Globally change the general search field name via yaml config**
+```yml
+SilverStripe\ORM\DataObject:
+  general_search_field_name: 'my_general_field_name'
+```
+
+**Customise the general search field name via yaml _or_ php config**
+```php
+use SilverStripe\ORM\DataObject;
+
+class MyDataObject extends DataObject
+{
+    private static string $general_search_field_name = 'my_general_field_name';
+}
+```
+
+#### Specify a search filter for general search
+
+By default, the general search will search across your fields using a [PartialMatchFilter](api:SilverStripe\ORM\Filters\PartialMatchFilter) regardless of what filters you have [specified for those fields](#specify-a-form-field-or-search-filter).
+
+You can configure this to be a specific filter class, or else disable the general search filter. Disabling the filter will result in the filters you have specified for each field being used when searching against that field in the general search.
+
+Like the general search field name, you can set this either globally or per class:
+
+**Globally change the general search filter via yaml config**
+```yml
+# use a specific filter
+SilverStripe\ORM\DataObject:
+  general_search_field_filter: 'SilverStripe\ORM\Filters\EndsWithFilter'
```

Comment:
> I don't think it should have precedence over configured filters

The way I see this is that the configured filters are _primarily_ for the specific search fields in the dropdown. They are separate from the functionality for the general search field. I would find it very strange if when I search "Big mac" in the general search if doesn't find someone who's last name is "Cormac" because someone decided last name should have a `StartsWithFilter`.

If developers do explicitly want their specific search field filters applied to the general search they can configure it to do that, but my instinct is that it's not intuitive to do so by default.
Might be another question for Cassie though.

Reply:
> If developers do explicitly want their specific search field filters applied to the general search

How would they do that?

Reply:
> You can configure this to be a specific filter class, or else disable the general search filter. Disabling the filter will result in the filters you have specified for each field being used when searching against that field in the general search.

Reply:
So to confirm, you CAN'T have say:
- PartialMatchFilter be the default on general search
- AND have PhoneNumber use ExactMatchFilter in the general search

If that's the case, this seems wrong. You shouldn't have to disable general search filter just to use the ExactMatchFilter on PhoneNumber

Also, if you disable the general search filter ... presumably the general search still works? What does it then essentially fall back to an ExactMatchFilter?

Reply:
No, you _can_ have that. Because `PartialMatchFilter` is the default for all searchable fields.

So the options are:
1. Always use a specific filter for _all_ fields when using general search. By default this is `PartialMatchFilter`
    - Whatever you configure for the specific fields will be used _only_ when using specific field search fields from the dropdown.
3. Always use the specific field filters when using general search. By default this is `PartialMatchFilter`
    - Any fields you have given a specific filter for the specific fields will apply to both the specific field search in the dropdown _and_ the search against that field in the general search.
    
    
What you cannot do is define a specific filter for two different fields, but only have one of those apply for the general search filter. We could add that as an enhancement further down the line, but I think it adds too much complexity of configuration to add that right now especially since it's an edge case.

Reply:
> if you disable the general search filter ... presumably the general search still works?

Yes. It will work, and use whatever filters are defined for the specific fields when searching against that column in the general search.

> What does it then essentially fall back to an ExactMatchFilter?

I don't understand this question.

Reply:
If someone disables the general search filter via config (default set in the PR is PartialMatchFilter) and the general search still functions, then the general search has to use a filter of some variety correct? Seems like it needs to be either a PartialMatchFilter or an ExactMatchFilter.  Even if it's not actually a Silverstripe filter, the database needs some form of logic to match values?

Reply:
> If someone disables the general search filter via config (default set in the PR is PartialMatchFilter) and the general search still functions, then the general search has to use a filter of some variety correct?

Yes.
The way it works it that it applies a filter for _each field_. By default this is always `PartialMatchFilter`. If you set this filter to a blank string in the config, it will instead check which filter is defined for that specific field (in `searchable_fields`) and use its filter. By default that will _also_ be `PartialMatchFilter`.

Reply:
OK thanks for clarifying.  Might be worth a bit of a wider discussion to see if we should have PartialMatchFilter as a default for the general filter then.

As things are, this is essentially an opt-out which is generally more of a breaking change (to be fair, the general search was very broken to begin with). The other option is to leave it blank and make it in opt-in to override what's already set on search fields. Things like phone numbers and SKU's might make more sense to have ExactMatchFilters.  To me the general and specific search should both respect the any configured filters

Reply:
@maxime-rainville any views on this?

Reply:
Note that `ExactMatchFilter` might give unexpected results with split terms, as per the documentation.

Reply:
Just summarising what I understood of the argument. You can choose which filter to use for each search field.

The proposed behaviour is to have the General field default to using partial match with the option to use a different filter via YML config. You can choose to customised the advanced filters on an individual basis, but that doesn't affect the general search.

The alternative we are considering here is that the general field should be driven by the filters of each individual advanced filter. e.g. I decide that the "Last name" filter will use "ends with" matching ... I search for "bob" ... all the field use partial matching except "last name" where it uses "ends with".

Presuming, I understood this correctly. I'm going to go with the first option: the general search field should use the same match strategy across all the fields and ignore the advanced filters.

In my mind, the General search field is completely disconnected from the advanced filters. It's the dumb field where people just throw a few words in and hope for a mildly sensible list of results. If you want to further filter your results, you can then choose to define an advanced filter that will apply on top of your general search.